src/lib.rs

```diff
@@ -246,6 +257,29 @@ macro_rules! aossoa {
             )*
         }
 
+        // Iterator ////////////////////////////////////////////////////////////
+
+        pub struct $iterator_struct_name<'a, T>
+            where T: 'a + $collection_trait_name<'a>
+        {
+            collection: &'a T,
+            index: usize,
+        }
+
+        impl<'a, T> Iterator for $iterator_struct_name<'a, T>
+            where T: 'a + $collection_trait_name<'a>
+        {
+            type Item = T::Ref;
+
+            fn next(&mut self) -> Option<Self::Item> {
+                let value = self.collection.get(self.index);
+                // TODO: We could probably reuse the index in the Ref type somehow
+                //       Possibly by making the Ref type the iterator
```

Comment:
I'm thinking maybe different implementations for AoS and SoA, but you're right, that TODO is a little to vague/big for an inline comment like that.